jobs/cjd_sample/src/main/java/com/google/samples/BatchOperationSample.java

```diff
@@ -0,0 +1,164 @@
+import com.google.api.services.jobs.v2.model.CreateJobRequest;
+import com.google.api.services.jobs.v2.model.Empty;
+import com.google.api.services.jobs.v2.model.Job;
+import com.google.api.services.jobs.v2.model.UpdateJobRequest;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Batch operations for job create, update and delete.
+ */
+public final class BatchOperationSample {
+
+  private static JobService jobService = JobServiceUtils.getJobService();
+
+  // [START batch_operation]
+
+  /**
+   * Batch operations for job create, update and delete.
+   */
+  public static void batchOperations() throws IOException {
+    String companyName = jobService.companies().create(
+        new Company().setDisplayName("BatchOperationSample")
+            .setDistributorCompanyId(
+                "BatchOperationSample:" + String.valueOf(new Random().nextLong()))).execute()
+        .getName();
+
+    List<Job> createdJobs = new ArrayList<>();
+
+    // Callback for batch create
+    JsonBatchCallback<Job> createCallback =
+        new JsonBatchCallback<Job>() {
+          @Override
+          public void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) {
+            System.out.println("Create Error Message: " + e.getMessage());
+          }
+
+          @Override
+          public void onSuccess(Job response, HttpHeaders responseHeaders) {
+            System.out.println("Create Job: " + response);
+            createdJobs.add(response);
+          }
+        };
+
+    Job softwareEngineerJob =
+        new Job()
+            .setCompanyName(companyName)
+            .setRequisitionId("123456")
+            .setJobTitle("Software Engineer")
+            .setApplicationUrls(Arrays.asList("http://careers.google.com"))
+            .setDescription("Design, develop, test, deploy, maintain and improve software.");
+    Job hardwareEngineerJob =
+        new Job()
+            .setCompanyName(companyName)
+            .setRequisitionId("1234567")
+            .setJobTitle("Hardware Engineer")
+            .setApplicationUrls(Arrays.asList("http://careers.google.com"))
+            .setDescription(
+                "Design prototype PCBs or modify existing board designs "
+                    + "to prototype new features or functions.");
+
+    // Creates batch request
+    BatchRequest batchCreate = jobService.batch();
+
+    // Queues create job request
+    jobService
+        .jobs()
+        .create(new CreateJobRequest().setJob(softwareEngineerJob))
+        .queue(batchCreate, createCallback);
+    jobService
+        .jobs()
+        .create(new CreateJobRequest().setJob(hardwareEngineerJob))
+        .queue(batchCreate, createCallback);
+
+    // Executes batch request
+    batchCreate.execute();
+
+    // Batch update jobs
+    List<Job> updatedJobs = new ArrayList<>();
+
+    JsonBatchCallback<Job> updateCallback =
```

Comment:
Can the create, update, and delete samples be in different methods, so they can be tested, copied, and verified independently?

I would also consider adding tags around them, so they can be linked to on devsite documentation using _github_include.

Reply:
We have another file 'CompanyAndJobCrudSample' which have the CRUD separated. This file is mainly targeting on batch operation and refereed by tag 'batch_operation' in our current documentation.

If you think it would better to put them into different methods, I might need to have a talk with my teammate to change our doc. What's your opinion?